xonsh/built_ins.py

```diff
@@ -637,15 +637,41 @@ def run_subproc(cmds, captured=True):
             hist.last_cmd_out = output
     if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
         raise CalledProcessError(hist.last_cmd_rtn, aliased_cmd, output=output)
-    if captured:
+    if captured=='stdout':
         return output
+    elif captured=='object':
+        return CompletedProcess(prev_proc, output)
 
 
-def subproc_captured(*cmds):
+class CompletedProcess:
```

Comment:
For what it is worth, I agree with @adqm's reasons.  The standard library is great, but it does have a lot of weird partially documented corner cases (eg readline) that should be avoided if possible.  In addition to the reasons that @adqm gave, I'll also add:

1. [`subprocess.CompletedProcess`](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess) was added in v3.5. So we'd either need to drop v3.4 support or have a substitute wrapper anyway.
2. Since it is new in the standard library, it is subject to change.
3. This is a relatively simple object, so the xontrol vs benefit of having our own slightly different version vs using an imperfect standard library object tilts in favor of having our own.
4. Duck typing is generally more important in Python than type checking.  I have struggled to think of a use case for this operator where "isinstance(p, subprocess.CompletedProcess)" is strictly more valuable than just having the same interface.  Are there cases that you can think of @piotrjurkiewicz - I am actually interested :)

Also, for what it is worth, I think that this should live in `xonsh.proc` (as @adqm suggested) and should probably have time stamps on it as well.

Anyway, I hope that helps.